### PR TITLE
Dan13436/git hooks

### PR DESCRIPTION
**Pull request by dan13436**
**Sample**

After executing the proper set up, Git should now automatically run a script every time you attempt to commit, checking if you are committing an API Key. If you are, then it will block your commit, if you are not, then the commit will proceed as normal.

_**Commits should not be blocked for any other reason.**_

**Summary**

After switching to this branch, make sure to read the read.me in the new folder and run the `git config core.hooksPath .githooks` command in your git terminal to confirm that the repo is using the correct githooks

**Tests**

Tested with Unity sample project with API key in editor and in project settings

**ArcGIS Maps SDK Version**

v 1.3

### CODE REVIEW

**Comment by Jade-JadeH** [comment]
huh, still being blocked after removing API keys.... 
image

**Comment by mmgaw113** [comment]
I get the same thing that jade does. It won't let you commit after you remove the API Key

**Comment by dan13436** [comment]
> huh, still being blocked after removing API keys.... image

It looks like the terminal is not recognizing python, based on the first line of output
`Python was not found; run without arguments to install from Microsoft Store`

I noted installing python in the readme but perhaps it will only recognize it through the Microsoft store? If you just type `python` into the terminal of your repo it should take you to the page where you can install it. If it works then please let me know so I could clarify that in the readme!

**Comment by dan13436** [comment]
> I get the same thing that jade does. It won't let you commit after you remove the API Key

Might be the same issue as Jade, will investigate

**Comment by mmgaw113** [comment]
Downloading python from the microsoft store does fix this, however you might want to research why it doesn't work when you already have python downloaded not through the microsoft store. That or find an alternative that will make it so we don't need to download python.

Not to mention some people use mac's and don't have the microsoft store to go and download this from so not sure what the solution is on there. Probably command line installation...

**Review by avr757 — APPROVED** [review]
after testing with @dan13436 on mac, we confirmed that its working.

**Review by NaumanIsGISMan — COMMENTED** [review]
I would try replacing calls to `python3` with `py`, the reason being that `py` is a utility function that should correctly id which version of python is installed on a windows machine:

https://stackoverflow.com/questions/50896496/what-is-the-difference-between-py-and-python-in-the-windows-terminal

This should help resolve the issue of needing to install the windows store version of python, but you definitely should try this change on mac as well as I'm not sure if that same utility will work on mac. 

In the main game engine repos, there are helpers that can detect the current platform, you might be able to port it from there from js into python.

You could also just try using `python` instead of `py`, I might try this first

File: .githooks/pythonScripts/check_api_key.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import re
+import argparse
+
+#-------------------------------------------------------------------------------
+# Global Variables
+#-------------------------------------------------------------------------------
+
+content = []
+
+unity_apiKey_argument_regex = r"apiKey[\s]*\:[\s]*([\sa-zA-Z0-9_\"\'\-]*)"
```
Comment (ZackAllen) [inline .githooks/pythonScripts/check_api_key.py:33]: This doesn't actually check for keys, you should be checking for the AAPK string

**Review by ZackAllen — CHANGES_REQUESTED** [review]
Commit history should be cleaned up to remove the test commits that were pushed. Force reset to the commit before and force push it up.

File: .githooks/pythonScripts/check_api_key.py (hunk 1 above)
Comment (dan13436) [reply .githooks/pythonScripts/check_api_key.py:33]: Yup! That's checked on line 65

**Review by NaumanIsGISMan — APPROVED** [review]

**Review by ZackAllen — APPROVED** [review]

**Pull request merged**